src/sentry/relay/config/metric_extraction.py

```diff
@@ -113,14 +111,19 @@ def get_metric_extraction_config(
     sentry_sdk.set_tag("organization_id", project.organization_id)
 
     with sentry_sdk.start_span(op="get_on_demand_metric_specs"):
-        alert_specs, widget_specs = get_on_demand_metric_specs(timeout, project)
+        on_demand_specs = run_experimental_config_builder(get_on_demand_metric_specs, project)
```

Comment:
Optional suggestion for ergonomics: You could add an optional argument for a default that is used instead of `None` when the experimental builder fails. Then, pass `([], [])` or `(None, None)` as default. This would make the signature larger, however.